test/test_transforms.py

```diff
@@ -305,5 +305,120 @@ def _test_librosa_consistency_helper(n_fft, hop_length, power, n_mels, n_mfcc, s
         _test_librosa_consistency_helper(**kwargs3)
 
 
+import pytest
+import numpy as np
+import torchaudio.functional as F
+xfail = pytest.mark.xfail
+
+
+@pytest.mark.parametrize('fft_length', [512])
+@pytest.mark.parametrize('hop_length', [256])
+@pytest.mark.parametrize('waveform', [
+    (torch.randn(1, 100000)),
+    (torch.randn(1, 2, 100000)),
+    pytest.param(torch.randn(1, 100), marks=xfail(raises=RuntimeError)),
+])
+@pytest.mark.parametrize('pad_mode', [
+    # 'constant',
+    'reflect',
+])
+def test_stft(waveform, fft_length, hop_length, pad_mode):
+    """
+    Test STFT for multi-channel signals.
+
+    Padding: Value in having padding outside of torch.stft?
+    """
+    pad = fft_length // 2
+    window = torch.hann_window(fft_length)
+    complex_spec = F.stft(waveform,
+                          fft_length=fft_length,
+                          hop_length=hop_length,
+                          window=window,
+                          pad_mode=pad_mode)
+    mag_spec, phase_spec = F.magphase(complex_spec)
+
+    # == Test shape
+    expected_size = list(waveform.size()[:-1])
+    expected_size += [fft_length // 2 + 1, test.common_utils._num_stft_bins(
+        waveform.size(-1), fft_length, hop_length, pad), 2]
+    assert complex_spec.dim() == waveform.dim() + 2
+    assert complex_spec.size() == torch.Size(expected_size)
+
+    # == Test values
+    fft_config = dict(n_fft=fft_length, hop_length=hop_length, pad_mode=pad_mode)
+    # note that librosa *automatically* pad with fft_length // 2.
+    expected_complex_spec = np.apply_along_axis(librosa.stft, -1,
```

Comment:
could we not import numpy? like have a for loop instead of np.apply_along_axis, torch.allclose for np.allclose, math.pi for np.pi, and math.ceil for np.ceil

Reply:
that's valid, but also we can think of some relaxation as long as we're import `librosa`, which will definitely rely on `numpy`.